src/memkind_memtier.c

```diff
@@ -742,6 +768,10 @@ memtier_builder_construct_memtier_memory(struct memtier_builder *builder)
 
 MEMKIND_EXPORT void memtier_delete_memtier_memory(struct memtier_memory *memory)
 {
+    if (memory->get_kind == memtier_policy_data_movement_get_kind) {
+        mtt_allocator_destroy(&memory->mtt_allocator);
```

Comment:
g_mtt_allocator is still pointing to the destroyed allocator... is it the desired behavior?